src/mame/drivers/tim011.cpp

```diff
@@ -92,14 +99,23 @@ void tim011_state::video_start()
 
 uint32_t tim011_state::screen_update_tim011(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
 {
+	u16 addr = 0;
+	for (u8 x = 0; x < 512/4; x++)
+	{
+		for (u16 y = 0; y < 256; y++)
+		{
+			int horpos = x*4;
+			u8 code = m_vram[addr++];
+			u8 realy = (y - m_scroll) & 0xff;
+			bitmap.pix(realy, horpos++) =  (code >> 0) & 0x03;
+			bitmap.pix(realy, horpos++) =  (code >> 2) & 0x03;
+			bitmap.pix(realy, horpos++) =  (code >> 4) & 0x03;
+			bitmap.pix(realy, horpos++) =  (code >> 6) & 0x03;
+		}
+	}
```

Comment:
You should restrict drawing to the clipping rectangle.  It’s particularly easy to do vertically.

Reply:
It actually always draw all 256 lines, just starts from different line to implement scrolling effect.

Reply:
I know it will do that the way it is now, but that’s undesirable as it will cause problems for partial updates (e.g. for mid-frame register changes).